src/services/expenses.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+let expenses = [];
+
+const getFilteredExpenses = (data) => {
+  let filteredExpenses = expenses;
+
+  const { userId, categories, from, to } = data;
+
+  if (userId) {
+    filteredExpenses = filteredExpenses.filter(
+      (expense) => expense.userId === Number(userId)
+    );
+  }
+
+  if (categories) {
+    filteredExpenses = filteredExpenses.filter(
+      (expense) => expense.category === categories
+    );
+  }
+
+  if (from && to) {
+    filteredExpenses = filteredExpenses.filter((expense) => {
+      return expense.spentAt >= from && expense.spentAt <= to;
+    });
+  }
+
+  return filteredExpenses;
+};
+
+const getById = (expenseId) =>
+  expenses.find((expense) => expense.id === Number(expenseId)) || null;
+
+const create = (data) => {
+  const { title, amount, spentAt, category, userId, note } = data;
+
+  const newExpense = {
+    id: expenses.length + 1,
```

Comment:
Instead of using the users' length to generate the new user ID, use a variable to store the next available ID and increment it every time a new user is created. This will avoid potential ID conflicts when users are removed.
Make a function for it or use `uuid` lib for it